src/components/diff-view.jsx

```diff
@@ -28,24 +28,37 @@ import SideBySideRawVersions from './side-by-side-raw-versions';
  * @param {DiffViewProps} props
  */
 export default class DiffView extends React.Component {
+  static getDerivedStateFromProps (props, state) {
```

Comment:
It took me some time to realize why `_loadingDiff` and  `previousDiff` were added to `diff-view` until I learned that gDSFP doesn't provide access to previous props! I saw some examples where the arguments for gDSFP were named `nextProps` and `prevState`. Do you think that would help clarify here (and any other place we use gDSFP)? Probably would've just helped in upfront costs for those not familiar with the new lifecycle method (like me), but adds more noise in the long-run.

Reply:
> Do you think that would help clarify here

Personally, no — for someone new to React in v16, I think that is carrying over old thinking and makes what’s happening *more* confusing.

I think It’s really important here to keep in mind that what you’re transforming are not props or state that are necessarily different from what came before. The reason they changed the name and made it static was to get *away* from people thinking about this function as dealing with transitions or “next” vs. “previous.” This function might get run multiple times with the same arguments and at arbitrary times; it’s no longer part of a component’s *lifecycle* and is meant to be an idempotent transformation (that is, you should be able to run it over and over and over again at any point in time, regardless of what’s happened elsewhere or in-between, and reliably get the same results).

Reply:
Ah, makes sense. Thanks for the explanation. Go ahead to hit all those green buttons =)